[BUG][CLOJURE] Clojure fails with java.lang.ClassCastException

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
  Expected: a Plaid client is produced

  Actual output:
```
Exception in thread "main" java.lang.RuntimeException: Could not process model 'ProfileGetResponse'.Please make sure that your schema is correct!
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:540)
        at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:1226)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:535)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class io.swagger.v3.oas.models.media.Schema (java.lang.Boolean is in module java.base of loader 'bootstrap'; io.swagger.v3.oas.models.media.Schema is in unnamed module of loader 'app')
        at org.openapitools.codegen.languages.ClojureClientCodegen.getTypeDeclaration(ClojureClientCodegen.java:174)
        at org.openapitools.codegen.DefaultCodegen.fromProperty(DefaultCodegen.java:4130)
        at org.openapitools.codegen.DefaultCodegen.fromProperty(DefaultCodegen.java:3931)
        at org.openapitools.codegen.DefaultCodegen.addParentContainer(DefaultCodegen.java:5838)
        at org.openapitools.codegen.DefaultCodegen.addAdditionPropertiesToCodeGenModel(DefaultCodegen.java:3722)
        at org.openapitools.codegen.DefaultCodegen.updateModelForObject(DefaultCodegen.java:2966)
        at org.openapitools.codegen.DefaultCodegen.fromModel(DefaultCodegen.java:3188)
        at org.openapitools.codegen.languages.ClojureClientCodegen.fromModel(ClojureClientCodegen.java:205)
        at org.openapitools.codegen.DefaultGenerator.processModels(DefaultGenerator.java:1681)
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:535)
        ... 4 more
```

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

The output is above and command to generate the output is below.

##### openapi-generator version

v7.4.0

##### OpenAPI declaration file content or url

[https://github.com/plaid/plaid-openapi/raw/master/2020-09-14.yml](https://github.com/plaid/plaid-openapi/raw/master/2020-09-14.yml)

##### Generation Details

`openapi-generator-cli generate -g clojure -i 2020-09-14.yml -o build/generated-clojure`

##### Steps to reproduce

1. Clone [plaid-openapi](https://github.com/plaid/plaid-openapi/tree/master)
2. Install OpenAPI Generator CLI v7.4.0
3. Run the above command (in "Generation Details")

##### Suggest a fix

from [ClojureClientCodegen.java#L174](https://github.com/OpenAPITools/openapi-generator/blob/1bfe800b9fdb74d31a41d7e7811ae1c3fbe7acd7/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ClojureClientCodegen.java#L174):
```java
Schema inner = (Schema) p.getAdditionalProperties();
```
This always suggests that getAdditionalProperties always returns a Schema object when it can also return a Boolean. Extra handling for the types might be needed.

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ClojureClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ClojureClientCodegen.java
@@ -171,9 +171,13 @@ public class ClojureClientCodegen extends DefaultCodegen implements CodegenConfi
 
             return "(s/coll-of " + getTypeDeclaration(inner) + ")";
         } else if (ModelUtils.isMapSchema(p)) {
-            Schema inner = (Schema) p.getAdditionalProperties();
-
-            return "(s/map-of string? " + getTypeDeclaration(inner) + ")";
+            Object additionalProperties = p.getAdditionalProperties();
+            // additionalProperties is either a Schema or a Boolean
+            if (additionalProperties instanceof Schema) {
+                Schema inner = (Schema) additionalProperties;
+                return "(s/map-of string? " + getTypeDeclaration(inner) + ")";
+            }
+            return "(s/map-of string? s/any?)";
         }
 
         // If it's a type we defined, we want to append the spec suffix

